docs/bsp.md

```diff
@@ -1586,27 +1586,35 @@ This is an example of a BSP connection file:
 
 #### Default Locations for BSP Connection Files
 
-A BSP connection file can be located in a number of locations:
-
-|        | Unix                                   | Windows                                   |
-|--------|----------------------------------------|-------------------------------------------|
-| System | `$XDG_DATA_DIRS/bsp/`                  | `%PROGRAMDATA%/bsp/`                      |
-| User   | `<workspace-dir>/.bsp/`                | `<workspace-dir>/.bsp/`                   |
+A BSP connection file can be located in a number of locations. BSP connection files
+may be located in the project workspace, or for bsp servers installed locally, in a system-specific
+directory in the user's data system's data directories:
+
+|           | Unix + Mac                                         | Windows                    |
+|-----------|----------------------------------------------------|----------------------------|
+| Workspace | `<workspace-dir>/.bsp/`                            | `<workspace-dir>\.bsp`     |
+| User      | `$XDG_DATA_HOME/bsp`                               | `%LOCALAPPDATA%\bsp`       |
+|           | `$HOME/Library/Application Support/bsp` (Mac only) |                            |
+| System    | `$XDG_DATA_DIRS/bsp/`                              | `%PROGRAMDATA%\bsp`        |
+|           | `/Library/Application Support/bsp` (Mac only)      |                            |
 
 Note that:
 
 1. `<workspace-dir>` refers to the workspace base directory.
-1. `$XDG_DATA_DIRS` is defined by the [XDG Base Directory
+1. `$XDG_DATA_HOME` and `$XDG_DATA_DIRS` are defined by the [XDG Base Directory
 Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.6.html)
-1. `%PROGRAMDATA%` is defined by the [Windows Documentation](https://docs.microsoft.com/en-us/windows-hardware/customize/desktop/unattend/microsoft-windows-shell-setup-folderlocations-programdata)
-
-The user location has always higher priority than the system location, so if
-a client finds a BSP connection file that meets its criteria inside a user
-location it must pick it over other BSP connection files in the system
+1. `%LOCALAPPDATA%` and `%PROGRAMDATA%` are defined by the 
+[Windows Documentation](https://docs.microsoft.com/en-gb/windows/desktop/shell/csidl) 
+(see also: [Default Known Folders](https://docs.microsoft.com/en-gb/windows/desktop/shell/knownfolderid))
+1. on Macs, both standard macOS and Unix directories are supported
+
+The workspace location always has higher priority than the user or system location, so if
```

Comment:
Unrelated to this PR since the paragraph was there before, but still want to comment that I'm on the fence about specifying this requirement here. Imagine I have a workspace with two different build tools (npm + sbt) and one build server is installed user-wide and other is project-specific. I think the client should be free to interpret the priority as it pleases.

Reply:
I’ll act on this in an independent PR, I’m batching small changes I want to add to the protocol.